src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -242,6 +245,17 @@ export const TradeConfirm = () => {
     [sellTradeId, trade],
   )
 
+  const chainAdapterManager = getChainAdapterManager()
+  const failureAmounts = useMemo(
+    () =>
+      failureAmountsCryptoPrecision?.map(({ buyAmountCryptoBaseUnit, asset }) => ({
+        amount: fromBaseUnit(buyAmountCryptoBaseUnit, asset.precision),
+        symbol: asset.symbol,
+        chainName: chainAdapterManager.get(asset.chainId)?.getDisplayName(),
```

Comment:
Are the intermediary, potentially failing trades guaranteed to be for an asset/chain we support? If so, can non-null assert since the adapter is guaranteed to be defined

Reply:
It's not guaranteed unfortunately, lifi supports about 3x the number of assets we do

Reply:
Does it mean the `chainName` would come as `undefined` in case of an intermediary from a chain we don't support?